contracts/BC_fusion/BSCGovernor.sol

```diff
@@ -0,0 +1,381 @@
+    error GovernorPaused();
+    // @notice signature: 0x286300de
+    error OnlyGovernorProtector();
+    // @notice signature: 0xb1d02c3d
+    error InBlackList();
+    // @notice signature: 0x867f3ee5
+    error OneLiveProposalPerProposer();
+
+    /*----------------- events -----------------*/
+    event Paused();
+    event Resumed();
+    event BlackListed(address indexed target);
+    event UnBlackListed(address indexed target);
+
+    /*----------------- storage -----------------*/
+    // target contract => is whitelisted for governance
+    mapping(address => bool) public whitelistTargets;
+
+    bool public proposeStarted;
+    bool public paused;
+
+    address public governorProtector;
+    mapping(address => bool) public blackList;
+
+    // @notice The latest proposal for each proposer
+    mapping(address => uint256) public latestProposalIds;
+
+    /*----------------- modifier -----------------*/
+    modifier whenNotPaused() {
+        if (paused) revert GovernorPaused();
+        _;
+    }
+
+    modifier onlyGovernorProtector() {
+        if (msg.sender != governorProtector) revert OnlyGovernorProtector();
+        _;
+    }
+
+    modifier notInBlackList() {
+        if (blackList[msg.sender]) revert InBlackList();
+        _;
+    }
+
+    /*----------------- init -----------------*/
+    function initialize() external initializer onlyCoinbase onlyZeroGasPrice {
+        __Governor_init("BSCGovernor");
+        __GovernorSettings_init(INIT_VOTING_DELAY, INIT_VOTING_PERIOD, INIT_PROPOSAL_THRESHOLD);
+        __GovernorCompatibilityBravo_init();
+        __GovernorVotes_init(IVotesUpgradeable(GOV_TOKEN_ADDR));
+        __GovernorTimelockControl_init(TimelockControllerUpgradeable(payable(TIMELOCK_ADDR)));
+        __GovernorVotesQuorumFraction_init(INIT_QUORUM_NUMERATOR);
+        __GovernorPreventLateQuorum_init(INIT_MIN_PERIOD_AFTER_QUORUM);
+
+        // BSCGovernor => Timelock => GovHub => system contracts
+        whitelistTargets[GOV_HUB_ADDR] = true;
+
+        // TODO
+        // Different address will be set depending on the environment
+        governorProtector = address(0xdEaD);
+    }
+
+    /*----------------- onlyGovernorProtector -----------------*/
+    /**
+     * @dev Pause the whole system in emergency
+     */
+    function pause() external onlyGovernorProtector {
+        paused = true;
+        emit Paused();
+    }
+
+    /**
+     * @dev Resume the whole system
+     */
+    function resume() external onlyGovernorProtector {
+        paused = false;
+        emit Resumed();
+    }
+
+    /**
+     * @dev Add an address to the black list
+     */
+    function addToBlackList(address account) external onlyGovernorProtector {
+        blackList[account] = true;
+        emit BlackListed(account);
+    }
+
+    /**
+     * @dev Remove an address from the black list
+     */
+    function removeFromBlackList(address account) external onlyGovernorProtector {
+        blackList[account] = false;
```

Comment:
How about remove it from the map to save storage?

Reply:
Done